src/hal/src/device.rs

```diff
@@ -734,6 +734,38 @@ pub trait Device<B: Backend>: fmt::Debug + Any + Send + Sync {
         flags: query::ResultFlags,
     ) -> Result<bool, WaitError>;
 
+    /// Create an acceleration structure object.
+    unsafe fn create_acceleration_structure(
+        &self,
+        desc: &acceleration_structure::AccelerationStructureDesc<B>,
+    ) -> Result<B::AccelerationStructure, OutOfMemory>;
+
+    /// Destroy an acceleration structure object.
+    unsafe fn destroy_acceleration_structure(
+        &self,
+        acceleration_structure: B::AccelerationStructure,
+    );
+
+    /// Get the size requirements for the buffers needed to build an acceleration structure.
+    unsafe fn get_acceleration_structure_build_requirements(
+        &self,
+        // used for host operations
+        // build_type: acceleration_structure::HostOrDevice,
+        build_info: &acceleration_structure::AccelerationStructureGeometryDesc<B>,
+        // must be a parallel array to `build_info.geometries` containing the primitive counts for each geometry.
+        max_primitives_count: &[u32],
+    ) -> acceleration_structure::AccelerationStructureSizeRequirements;
+
+    // TODO these are used for host operations, which DX12 does not support.
+    // - build_acceleration_structures
+    // - copy_acceleration_structure
+    // - copy_acceleration_structure_to_memory
+    // - copy_memory_to_acceleration_structure
+    // - write_acceleration_structures_properties
+
+    // TODO for checking if a serialized blob is valid with the current driver version. DX12 docs say this is for PIX tooling and building a as from scratch is "likely to be faster than loading one from disk" (https://docs.microsoft.com/en-us/windows/win32/api/d3d12/ne-d3d12-d3d12_raytracing_acceleration_structure_copy_mode). The Vulkan spec/literature doesn't mention perf implications of serialization.
+    // - get_device_acceleration_structure_compatibility
```

Comment:
This function operates basically the same in Vulkan (VkGetDeviceAccelerationStructureCompatibility) and DX (CheckDriverMatchingIdentifier). Both take a pointer to the serialized data and read two GUIDs (32 bytes) to see if it matches the current driver (DX can signal slightly more info than VK).

On the comment in `AccelerationStructureCopyMode`, I wonder if serializing an acceleration structure is something we need to expose as a part of the API. I can't find any literature on why one would want to do this in Vulkan, but as said [mentioned link](https://docs.microsoft.com/en-us/windows/win32/api/d3d12/ne-d3d12-d3d12_raytracing_acceleration_structure_copy_mode), DX mainly uses serialization for PIX and not to save on build time. If this is also true for Vulkan (like RenderDoc wants it or something), we may be able to leave this off the API and handle the debug tooling case as a backend implementation detail (i.e. I don't think people are writing debug tooling targetting the gfx api yet)

Reply:
Wouldn't you want a ray-traced scene to load faster if the data like this is loaded from disk?
I'd be totally fine just postponing any serialization stuff if you don't consider it useful.

Reply:
Do you know of a vulkan resource where we can ask about the perf implications of serialization (or just general "best practices")? The DX docs (for `D3D12_RAYTRACING_ACCELERATION_STRUCTURE_COPY_MODE_DESERIALIZE`) seem to imply this isn't a perf win:

> This mode is only intended for tools such as PIX, though nothing stops any app from using it, but this mode is only permitted when developer mode is enabled in the OS. This copy operation is not intended to be used for caching acceleration structures, because running a full acceleration structure build is likely to be faster than loading one from disk.

Reply:
Khronos Slack would be the best place to ask